compiler/tests/evm_contracts.rs

```diff
@@ -293,6 +293,29 @@ fn test_revert() {
     })
 }
 
+#[test]
+fn test_static_string() {
+    with_executor(&|mut executor| {
+        let harness = deploy_contract(&mut executor, "static_string.fe", "Foo", vec![]);
+
+        harness.test_function(&mut executor, "return_string", vec![string_token("foo")], Some(string_token("foo")));
+
+        harness.test_function(&mut executor, "foo_equal_foo", vec![], Some(bool_token(true)));
+        harness.test_function(&mut executor, "foo_unequal_foo", vec![], Some(bool_token(false)));
+
+        harness.test_function(&mut executor, "passed_equal_foo", vec![string_token("foo")], Some(bool_token(true)));
+
```

Comment:
The following three tests fail (first one just hangs the test suite) and they make me question if the generated YUL code actually does what I hope it does. The surprising equality checks may be due because we actually haven't defined how equality should be checked for this data type but the hanging of the test suite for when I want to return a static string is even more surprising.